File: php/features.php
<?php
declare(strict_types=1);

// RedditStocks SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RedditStocksFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RedditStocksBaseFeature();
            case "test":
                return new RedditStocksTestFeature();
            default:
                return new RedditStocksBaseFeature();
        }
    }
}

File: py/features.py
# RedditStocks SDK feature factory

from feature.base_feature import RedditStocksBaseFeature
from feature.test_feature import RedditStocksTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RedditStocksBaseFeature(),
        "test": lambda: RedditStocksTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
